### PR TITLE
Libxdp documentation + functions

### PR DESCRIPTION
**Pull request by DaFray31**
Libxdp documentation

**Missing** : 
- [x] Errors return by functions
- [x] All AF_XDP functions

### CODE REVIEW

**Comment by dylandreimerink** [comment]
This is awesome :rocket:, thanks for working on it

**Comment by DaFray31** [comment]
Should be good now.
I hope, I haven't forgotten anything

File: data/libxdp-functions.yaml
Comment (dylandreimerink) [inline data/libxdp-functions.yaml:1]: What is the reason for adding this data file? We typically use these data files in conjunction with some tooling to generate parts of the docs. They are not a strict requirement.

File: docs/ebpf-library/libxdp/libxdp.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
+# LIBXDP
+
+libxdp is a light eBPF library who add 2 features for [XDP programs](../../linux/program-type/BPF_PROG_TYPE_XDP.md).
+
+- Load multiple programs on single network device using a "dispatcher program" thanks to `freplace`
+- Configuring [`AF_XDP`](/linux/concepts/af_xdp) and functions to read and write on theses sockets
```
Comment (dylandreimerink) [inline docs/ebpf-library/libxdp/libxdp.md:6]:
```suggestion
- Configuring [`AF_XDP`](../../linux/concepts/af_xdp) and functions to read and write on theses sockets
```

File: docs/ebpf-library/libxdp/libxdp.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
+# LIBXDP
+
+libxdp is a light eBPF library who add 2 features for [XDP programs](../../linux/program-type/BPF_PROG_TYPE_XDP.md).
+
+- Load multiple programs on single network device using a "dispatcher program" thanks to `freplace`
```
Comment (dylandreimerink) [inline docs/ebpf-library/libxdp/libxdp.md:5]:
```suggestion
- Load multiple programs on single network device using a "dispatcher program" thanks to [`freplace`](../../linux/program-type/BPF_PROG_TYPE_EXT.md)
```

File: docs/ebpf-library/libxdp/SUMMARY.md
Comment (dylandreimerink) [inline docs/ebpf-library/libxdp/SUMMARY.md:1]: I suggest changing the function names to the lowercase variants, so everything is more uniform

**Review by dylandreimerink — APPROVED** [review]
This is great! Thanks for contributing this.
Just a few small things

File: data/libxdp-functions.yaml
Comment (DaFray31) [reply data/libxdp-functions.yaml:1]: Not necessarily useful, but maybe in the future? (track update or permissions ?)

Comment (dylandreimerink) [reply data/libxdp-functions.yaml:1]: I am a bit afraid that the file might rot if its not actually in use. I will take it for now and see if we can use it for something.

**Pull request merged**